### DISCUSSION TITLE
possible issue: map file much larger than it should be

### DISCUSSION THREAD

**Discussion by gatech79** [mapsforge/mapsforge#1282] [General]
About a little over a year ago I noticed the us_south.map and alaska.map became
much larger, overnight.
Example:
region             2020 .map size       2022 .map size
-------------------    -----------------------        ------------------------
us_midwest       1007MB                 1278MB
us_northeast       634MB                   929MB
us_pacific            663MB                  5192MB  (this entry is suspicious)
us_south            1700MB                  2080MB
us_west              1340MB                 1634MB

This also shows up in the US "state" .map file for Alaska
Alaska has the smallest road network in the entire USA
yet is a much bigger .map file compared to any other state
by a huge margin.

Texas is about the same size of alaska and Texas has a much denser road network.
Yet Texas.map is 380MB while alaska.map is 3026MB. This might be a correct
result for some reason, but my intuition tells me something went wrong here.
The Alaska .map is perfectly usable other wise.

I use the mapsforge maps in several apps.
such as Cruiser, Amap. I'm not a developer, really just a user.

Kevin

**Comment by gatech79** [comment]
In the first line "us_south.map" should be "us_pacific.map"

**Reply by devemux86** [reply]
You can edit your posts. :slightly_smiling_face:

**Reply by gatech79** [reply]
thx.

**Comment by gatech79** [comment]
Dug into it a little bit.

Used BBBike to extract a mapsforge file of most of alaska using a simple rectangular boundary.
The resulting alaska.map is 132MB and is usable, but no coastline.

Down loaded alaska.map from https://www.openandromaps.org/en/downloads/usa_en 
There alaska.map is 1542MB and has a coastline. There map was rendered 23 days ago.

Alaska.map from http://download.mapsforge.org/ also has a coastline also
and was rendered 13 days ago. It's file size 3026MB
So perhaps it takes 2.8GB to include a detailed coastline of alaska.
Kevin

**Comment by gatech79** [comment]
I looked around for another land mass that has very few roads
but has a lot of coastline. The "far_eastern_fed_district.map" in Russia
has very few roads(Siberia), but a lot of coastline. It's 5.9GB. I image 5.5GB is
just coastline. Maybe it's just the cost of having a detailed coastline.
Kevin

**Comment by gatech79** [comment]
With more google searches, this problem was found by others five years ago.
Guess I'm late to the party. I won't pursue it any more as people with much more talent
than me are already aware that something is strange.

Looking at 
http://download.geofabrik.de/north-america.html
and
http://download.mapsforge.org/
the .pbf and .map are generally within about 20% of each other.

I found three cases that do not follow this pattern

                             pbf       map
us_pacific           128MB    5000MB
nunavet(canada)   95MB   1200MB
Russia                  3000MB   10021MB

Nunavet is the most interesting. That whole part of Canada is just
one big chunk of ice with hardly a road. Does have a lot of coastline and lakes.
Kevin

**Comment by gatech79** [comment]
I've played with the large maps quite a bit. I'm impressed.
What's amazing is that Cruiser still presents a very fast, crisp, interactive view
of the maps. Panning, zooming are just as fast a small maps.
Large maps do not seem to effect the usability at all.
I don't know that clever stuff is being done behind the scenes here,
but it works very well. Don't know if the entire map is being memory 
to the cpu or if the data is so well organized that file-seeks are always 
not very far away. But Cruiser handles it with no problem.

And phones today can take data of that size. So even with my low end
phone I have no problems Cruisers quick response with zooming and panning
on the map. Amazing!  While my usage patterns today I don't really
need details coastlines, that is not a restriction I would not apply to others.
I'm sure many users would like to be able to travel long distances, and at there destination
go canoeing, kayaking, boating and would like the detailed lakes, rivers, coastline.

Only reason I've added this comment is at first I was expecting a performance degradation
with large maps. But I have seen no evidence of this. So what you guys(girls) did
to keep high performance with large maps. Good job.

Kevin

**Reply by devemux86** [reply]
Cruiser GL (with Open GL [VTM](https://github.com/mapsforge/vtm)) is faster than Cruiser (with [Mapsforge](https://github.com/mapsforge/mapsforge)).

The maps are the same. Mapsforge format specification is described [here](https://github.com/mapsforge/mapsforge/blob/master/docs/Specification-Binary-Map-File.md).
